### PR TITLE
[CINFRA] Fix race when creating an Index in Rep2. reporting success too early

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

*When we create an index in Replication 2, we need to wait for the RaftIndex of the "isBuilding" removal, this is now happening. Also replace the complex waiting logic by a easier to understand future based approach.*

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket: [CINFRA-830](https://arangodb.atlassian.net/browse/CINFRA-830)
- [ ] Design document: 

[CINFRA-830]: https://arangodb.atlassian.net/browse/CINFRA-830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: arangod/Cluster/ClusterIndexMethods.cpp [hunk 1]
```diff
@@ -932,61 +854,66 @@ auto ensureIndexCoordinatorReplication2Inner(
...
-        break;
-      }
+    VPackBuilder resultBuilder;
+    {
+      VPackObjectBuilder guard(&resultBuilder);
+      // add all original values
+      resultBuilder.add(VPackObjectIterator(newIndexBuilder.slice()));
+      resultBuilder.add("isNewlyCreated", VPackValue(true));
     }
+    // Need to wait here, until ClusterInfo has updated to latest plan.
```
Comment (apetenchea) [inline arangod/Cluster/ClusterIndexMethods.cpp:903]: Are we waiting implicitly, as a consequence of the above callback?
Or, is this something that should be added in a separate PR?

Comment (mchacki) [reply arangod/Cluster/ClusterIndexMethods.cpp:903]: The comment is in the wrong line .... Will fix

**Review by apetenchea — APPROVED** [review]

**Comment by mchacki** [comment]
[90b094b](https://github.com/arangodb/arangodb/pull/20081/commits/90b094b9d597d56702db6556676a68c9dc0a93d3) had a green CI run, the later addition was just moving comments.
